ibm/service/cos/resource_ibm_cos_bucket.go

```diff
@@ -435,6 +435,13 @@ func ResourceIBMCOSBucket() *schema.Resource {
 				Default:     true,
 				Description: "COS buckets need to be empty before they can be deleted. force_delete option empty the bucket and delete it.",
 			},
+			"object_lock": {
+				Type:         schema.TypeBool,
+				Optional:     true,
+				RequiredWith: []string{"object_versioning"},
+				Default:      false,
```

Comment:
Do we need a default value since we told that its required with object_versioning. 
Can you test a import scenario of the simple bucket ?

We are not setting this value in ReadMethod?

Reply:
addressed the comment

Reply:
Object versioning is required with object lock . I did not get  "Do we need a default value since we told that its required with object_versioning."

Reply:
The second part I have addressed , added the code in the Read Method.

Reply:
addressed the comment